plugins/wazuh-core/public/services/server-security/types.ts

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { BehaviorSubject, Subject } from 'rxjs';
+
+export interface ServerSecurityPermission {
+  action: string;
+  resource: string;
+}
+
+export type ServerSecurityCombinedPermission =
+  | ServerSecurityPermission
+  | ServerSecurityPermission[];
+
+export type ServerSecurityCombinedPermissionWithFunction =
+  | ServerSecurityCombinedPermission
+  | ((props: any) => ServerSecurityPermission);
+
+export interface ServerSecuritySetupDeps {
+  auth$: Subject;
+  useDashboardSecurityAccount: () => {
+    administrator_requirements: string | null;
+  };
+  useLoadingLogo: () => { url: string; type: string };
+}
+
+export interface ServerSecuritySetupReturn {
+  hooks: {
+    useServerUserLogged: () => boolean;
+    useServerUserPermissions: () => any;
+    useServerUserPermissionsRequirements: (
+      permissions: ServerSecurityCombinedPermissionWithFunction,
+    ) => [ServerSecurityCombinedPermission, any];
+  };
+  hocs: {
+    withServerUserAuthorizationPrompt: (
+      permissions: ServerSecurityCombinedPermissionWithFunction | null,
+      otherPermissions: { isAdmininistrator: boolean | null },
+    ) => (WrappedComponent: React.Component) => React.ReactElement;
```

Comment:
```suggestion
    ) => (WrappedComponent: React.ElementType) => React.ReactElement;
```

Reply:
Solved in 12187ab1638412e0cc8155aecfd00cbe75a6253f.

Use `React.ComponentType` instead.